cnd/src/network/comit.rs

```diff
@@ -254,6 +211,49 @@ impl Comit {
         // We trust in libp2p to poll us.
         Poll::Pending
     }
+
+    fn remote_data_insert<T>(&mut self, shared_swap_id: SharedSwapId, value: T)
+    where
+        RemoteData: Set<T>,
+    {
+        let mut remote_data = self
+            .remote_data
+            .get(&shared_swap_id)
+            .cloned()
+            .unwrap_or_default();
+        remote_data.set(value);
+        self.remote_data.insert(shared_swap_id, remote_data);
+    }
+
+    fn finalize(&mut self, peer: PeerId, shared_swap_id: SharedSwapId) {
+        let local_swap_id = match self.swaps.get_local_swap_id(shared_swap_id) {
+            Some(id) => id,
+            None => return,
```

Comment:
:question: Should we log something if we try to finalize a swap and we don't have a reference to it?

Reply:
This function is called so many times, right now I'm not keen to add any tracing to it. May be worth further investigation (especially with your comment on this function below :)